src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs

```diff
@@ -2092,7 +2092,9 @@ private IEnumerable<TypeInferenceInfo> InferTypeInVariableComponentAssignment(Ex
                         }
                         else if (expr.IsKind(SyntaxKind.IdentifierName))
                         {
-                            elementNamesBuilder.Add(((IdentifierNameSyntax)expr).Identifier.ValueText);
+                            elementNamesBuilder.Add(string.IsNullOrWhiteSpace(((IdentifierNameSyntax)expr).Identifier.ValueText) ? 
+                                null : 
+                                ((IdentifierNameSyntax)expr).Identifier.ValueText);
```

Comment:
fwiw @jcouv .  i do find it odd that there is this restriction in the core API.  i.e. you allow null.  you allow whitespace.  you allow normal text.  but you basically don't allow the empty string.  seems... string to me.  But if you really want that, then this change seems ok :)

@chborl small nit.  could you just make this:

```c#
else if (expr is IdentifierName identifier) { ...
```

Then you don't need the multiple casts.

Reply:
@chborl since 'null' is allowed, this check should be:

`elementNamesBuilder.Add(name.Identifier.ValueText == "" ? null : name.Identifier.ValueText)`

Reply:
@CyrusNajmabadi I agree the restriction is a bit strange, but arguably this PR demonstrates why it is useful: it helps guide users to pass `null` rather than empty, when they want to represent missing names ;-)

Reply:
@jcouv asking since I saw Chery changed back from IsNullOrWhitespace to checking explicitly just "" string.

will null or whitespaces input will change any outcome of the API?

Reply:
Correct, you can create a tuple symbol with illegal names, but we block empty strings.

This check was added in PR https://github.com/dotnet/roslyn/pull/14958 
The most relevant [comment](https://github.com/dotnet/roslyn/pull/14958#discussion_r87488278) I could find is that "fields can't have blank names".

This was tested explicitly:
```C#
        [Fact]
        public void CreateTupleTypeSymbol2_BadNames()
        {
            var comp = CSharpCompilation.Create("test", references: new[] { MscorlibRef });

            ITypeSymbol intType = comp.GetSpecialType(SpecialType.System_Int32);

            // illegal C# identifiers and blank
            var tuple2 = comp.CreateTupleTypeSymbol(ImmutableArray.Create(intType, intType), ImmutableArray.Create("123", " "));
            Assert.Equal(new[] { "123", " " }, GetTupleElementNames(tuple2));

            // reserved identifiers
            var tuple3 = comp.CreateTupleTypeSymbol(ImmutableArray.Create(intType, intType), ImmutableArray.Create("return", "class"));
            Assert.Equal(new[] { "return", "class" }, GetTupleElementNames(tuple3));
        }
```